operators/src/plot/statistics.rs

```diff
@@ -0,0 +1,261 @@
+    fn query_processor(&self) -> Result<TypedPlotQueryProcessor> {
+        Ok(TypedPlotQueryProcessor::JSON(
+            StatisticsQueryProcessor {
+                rasters: self
+                    .raster_sources
+                    .iter()
+                    .map(|source| source.query_processor())
+                    .collect::<Result<Vec<_>>>()?,
+            }
+            .boxed(),
+        ))
+    }
+}
+
+/// A query processor that calculates the statistics about its inputs.
+pub struct StatisticsQueryProcessor {
+    rasters: Vec<TypedRasterQueryProcessor>,
+}
+
+impl PlotQueryProcessor for StatisticsQueryProcessor {
+    type PlotType = serde_json::Value;
+
+    fn plot_query<'a>(
+        &'a self,
+        query: QueryRectangle,
+        ctx: &'a dyn QueryContext,
+    ) -> BoxFuture<'a, Result<Self::PlotType>> {
+        let mut queries = Vec::with_capacity(self.rasters.len());
+        for (i, raster_processor) in self.rasters.iter().enumerate() {
+            queries.push(
+                call_on_generic_raster_processor!(raster_processor, processor => {
+                    processor.query(query, ctx)
+                             .map(move |r| r.map(|tile| (i, tile.convert::<f64>())))
+                             .boxed()
+                }),
+            );
+        }
+
+        let number_statistics = vec![NumberStatistics::default(); self.rasters.len()];
+
+        select_all(queries)
+            .fold(
+                Ok(number_statistics),
+                |number_statistics: Result<Vec<NumberStatistics>>, enumerated_raster_tile| async move {
+                    let mut number_statistics = number_statistics?;
+                    let (i, raster_tile) = enumerated_raster_tile?;
+
+                    process_raster(&mut number_statistics[i], &raster_tile);
+
+                    Ok(number_statistics)
+                },
+            )
+            .map(|number_statistics| {
+                let output: Vec<StatisticsOutput> = number_statistics?.iter().map(StatisticsOutput::from).collect();
+                serde_json::to_value(&output).map_err(Into::into)
+            })
+            .boxed()
+    }
+}
+
+#[allow(clippy::float_cmp)] // allow since NO DATA is a specific value
+fn process_raster(number_statistics: &mut NumberStatistics, raster_tile: &RasterTile2D<f64>) {
+    let no_data_value = raster_tile.grid_array.no_data_value;
+
+    if let Some(no_data_value) = no_data_value {
+        for &value in &raster_tile.grid_array.data {
+            if value == no_data_value {
+                number_statistics.add_no_data();
+            } else {
+                number_statistics.add(value);
+            }
+        }
+    } else {
+        for &value in &raster_tile.grid_array.data {
+            number_statistics.add(value);
+        }
+    }
+}
+
+/// The statistics summary output type for each raster input
+#[derive(Debug, Clone, Copy, Serialize, Deserialize)]
+struct StatisticsOutput {
+    pub count: usize,
```

Comment:
pixel_count would be a more descriptive name

Reply:
4b963e9